packages/@lightningjs/ui-components/src/components/Column/Column.test.js

```diff
@@ -399,138 +431,137 @@ describe('Column', () => {
         testRenderer.keyPress('Down');
-        testRenderer.update();
-        setTimeout(() => {
-          expect(column.items.length).toBe(9);
-        }, 17);
+        expect(column.items.length).toBe(10);
       });
     });
 
     describe('with column height < items', () => {
-      const waitForSmooth = column => {
-        return completeAnimation(column.Items, 'y');
-      };
+      beforeEach(async () => {
+        [column, testRenderer] = createColumn(
+          {
+            h: 400
+          },
+          { spyOnMethods: ['_update'] }
+        );
 
-      beforeEach(() => {
-        column.h = 400;
         expect(column._Items.y).toBe(0);
       });
 
       describe('and scrollIndex = 0', () => {
         it('should scroll down', async () => {
           testRenderer.keyPress('Down');
-          await waitForSmooth(column);
+          await completeAnimation(column._Items, 'y');
           expect(column._Items.y).toBe(-100);
         });
 
         it('should scroll up', async () => {
           testRenderer.keyPress('Down');
-          await waitForSmooth(column);
+          await completeAnimation(column._Items, 'y');
           expect(column._Items.y).toBe(-100);
 
           testRenderer.keyPress('Up');
-          await waitForSmooth(column);
+          await completeAnimation(column._Items, 'y');
           expect(column._Items.y).toBe(0);
         });
       });
 
       describe('and scrollIndex = 2', () => {
-        beforeEach(() => {
-          column.items = items.concat(items);
-          column.scrollIndex = 2;
-          testRenderer.update();
+        beforeEach(async () => {
+          [column, testRenderer] = createColumn(
+            {
+              items: items.concat(items),
+              scrollIndex: 2,
+              h: 400
+            },
+            { spyOnMethods: ['_update'] }
+          );
+          await column.__updateSpyPromise;
+
+          expect(column.scrollIndex).toBe(2);
+          expect(column.items.length).toBe(10);
         });
 
-        xit('should render correctly', () => {
+        it('should render correctly', () => {
           expect(column.items[0].y).toBe(0);
-          expect(column.items[1].y).toBe(100);
+          expect(column.items[1].h).toBe(80);
         });
 
         it('should not scroll until past the mid point', () => {
           const [item] = column.items;
-          testRenderer.keyPress('Down');
-          testRenderer.update();
           expect(item.y).toBe(0);
         });
 
-        xit('should scroll down', async () => {
+        it('should scroll down', async () => {
           testRenderer.keyPress('Down');
           testRenderer.keyPress('Down');
           testRenderer.keyPress('Down');
-          await waitForSmooth(column);
+          await completeAnimation(column._Items, 'y');
           expect(column._Items.y).toBe(-column.items[1].y);
         });
 
-        it('should scroll up', () => {
+        // timing out still
```

Comment:
can't figure out why this test keeps timing out. If anyone as thoughts let me know.

Reply:
It's timing out because the `scrollIndex` is set to `2` in the `describe` block that it's within. That means that no scrolling should occur until navigating to items after the item at index 2. In this test `testRenderer.keyPress('Down')` is only being invoked once, so focus is just going from index 0 ---> 1 and wont' cause any scrolling to occur. Calling that function 3 times will result in scrolling one position on the third call as focus goes from item at index 2 --> 3.

```js
it('should scroll up', async () => {
  testRenderer.keyPress('Down'); // 0 --> 1, no scroll
  testRenderer.keyPress('Down'); // 1 --> 2, no scroll
  testRenderer.keyPress('Down'); // 2 --> 3, passed scrollIndex 2, scroll
  await completeAnimation(column._Items, 'y');
  expect(column._Items.y).toBe(-100); // testing to make sure scrolls down
  testRenderer.keyPress('Up');
  await completeAnimation(column._Items, 'y');
  expect(column._Items.y).toBe(0);
});
```


I think a lot of the tests nested in `describe('and scrollIndex = someNumber'')` could be combined or removed, but not necessary to tackle that here.